packages/graph/hash_graph/lib/graph/src/ontology/property_type.rs

```diff
@@ -26,7 +31,243 @@ impl QueryRecord for PropertyType {
+            PropertyTypeQueryToken::VersionedUri => PropertyTypeQueryPath::VersionedUri,
+            PropertyTypeQueryToken::Version => PropertyTypeQueryPath::Version,
+            PropertyTypeQueryToken::Title => PropertyTypeQueryPath::Title,
+            PropertyTypeQueryToken::Description => PropertyTypeQueryPath::Description,
+            PropertyTypeQueryToken::DataTypes => {
+                seq.next_element::<Selector>()?
+                    .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+                self.position += 1;
+
+                let data_type_query_path =
+                    DataTypeQueryPathVisitor::new(self.position).visit_seq(seq)?;
+
+                PropertyTypeQueryPath::DataTypes(data_type_query_path)
+            }
+            PropertyTypeQueryToken::PropertyTypes => {
+                seq.next_element::<Selector>()?
+                    .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+                self.position += 1;
+
+                let property_type_query_path = Self::new(self.position).visit_seq(seq)?;
+
+                PropertyTypeQueryPath::PropertyTypes(Box::new(property_type_query_path))
+            }
+        })
+    }
+}
+impl<'de: 'k, 'k> Deserialize<'de> for PropertyTypeQueryPath<'k> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_seq(PropertyTypeQueryPathVisitor::new(0))
+    }
+}
+
+pub enum Selector {
+    Wildcard,
+}
+
+impl<'de> Deserialize<'de> for Selector {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct SelectorVisitor;
+
+        impl<'de> Visitor<'de> for SelectorVisitor {
+            type Value = Selector;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a wildcard (*)")
+            }
+
+            fn visit_str<E: de::Error>(self, v: &str) -> Result<Self::Value, E> {
+                match v {
+                    "*" => Ok(Selector::Wildcard),
+                    _ => Err(de::Error::invalid_value(Unexpected::Str(v), &self)),
+                }
+            }
+
+            fn visit_bytes<E: de::Error>(self, v: &[u8]) -> Result<Self::Value, E> {
+                match core::str::from_utf8(v) {
+                    Ok(s) => self.visit_str(s),
+                    Err(_) => Err(E::invalid_value(de::Unexpected::Bytes(v), &self)),
+                }
+            }
+        }
+
+        deserializer.deserialize_str(SelectorVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::ontology::test_utils::create_path;
+
+    fn convert_path(
+        segments: impl IntoIterator<Item = &'static str>,
+    ) -> PropertyTypeQueryPath<'static> {
+        PropertyTypeQueryPath::try_from(create_path(segments)).expect("Could not convert path")
```

Comment:
We agreed that panic messages should be lowercase right? If so can we update all of these

Reply:
I think I fixed it for every message with 8690ed797b6b84558d56174bdbc0a9c17a968077.